src/main/java/org/apache/beam/sdk/io/gcp/healthcare/DicomIO.java

```diff
@@ -0,0 +1,175 @@
+                if (pct.getAll()
+                        .keySet()
+                        .containsAll((Collection<?>) TupleTagList.of(METADATA).and(ERROR_MESSAGE))) {
+                    return new ReadStudyMetadata.Result(pct);
+                } else {
+                    throw new IllegalArgumentException(
+                            "The PCollection tuple must have the DicomIO.ReadStudyMetadata.METADATA "
+                                    + "and DicomIO.ReadStudyMetadata.ERROR_MESSAGE tuple tags");
+                }
+            }
+
+            private Result(PCollectionTuple pct) {
+                this.pct = pct;
+                this.readResponse = pct.get(METADATA);
+                this.failedReads = pct.get(ERROR_MESSAGE);
+            }
+
+            /**
+             * Gets failed reads.
+             *
+             * @return the failed reads
+             */
+            public PCollection<String> getFailedReads() {
+                return failedReads;
+            }
+
+            /**
+             * Gets resources.
+             *
+             * @return the resources
+             */
+            public PCollection<String> getReadResponse() {
+                return readResponse;
+            }
+
+            @Override
+            public Pipeline getPipeline() {
+                return this.pct.getPipeline();
+            }
+
+            @Override
+            public Map<TupleTag<?>, PValue> expand() {
+                return ImmutableMap.of(METADATA, readResponse);
+            }
+
+            @Override
+            public void finishSpecifyingOutput(
+                    String transformName, PInput input, PTransform<?, ?> transform) {}
+        }
+
+        /**
+         * DoFn to fetch the metadata of a study from a Dicom store based on it's location and study id.
+         */
+        static class FetchStudyMetadataFn extends DoFn<String, String> {
+
+            private HealthcareApiClient dicomStore;
+
+            /**
+             * Instantiate the healthcare client.
+             *
+             * @throws IOException
+             */
+            @Setup
+            public void instantiateHealthcareClient() throws IOException {
+                this.dicomStore = new HttpHealthcareApiClient();
+            }
+
+            /**
+             * Process The Pub/Sub message.
+             *
+             * @param context The input containing the pub/sub message
+             */
+            @ProcessElement
+            public void processElement(ProcessContext context) {
+                String dicomWebPath = context.element();
+                try {
+                    // TODO Change to non-blocking async calls
+                    String responseData = dicomStore.retrieveDicomStudyMetadata(dicomWebPath);
+                    context.output(METADATA, responseData);
+                } catch (Exception e) {
```

Comment:
Catching all exceptions is a bad practice: see https://stackoverflow.com/questions/2416316/why-is-the-catchexception-almost-always-a-bad-idea